Updated env example and readme (#14)
BreadchainCoop/eigenlayer-bls-local: README.md

   ```
   cp example.env .env
   ```
   Make sure you have filled in all the required variables. The environment variables are organized into several sections:

>    ### Mainnet/Holesky Configuration

Reviewer: ```suggestion
   ### Mainnet/Testnet Configuration
```

   Choose one set of addresses based on your target network:
   ```
   DELEGATION_MANAGER_ADDRESS=  # Address of the delegation manager contract
   STRATEGY_MANAGER_ADDRESS=    # Address of the strategy manager contract
   LST_CONTRACT_ADDRESS=        # Address of the LST contract
   LST_STRATEGY_ADDRESS=        # Address of the LST strategy contract
   BLS_SIGNATURE_CHECKER_ADDRESS= # Address of the BLS signature checker
   OPERATOR_STATE_RETRIEVER_ADDRESS= # Address of the operator state retriever
   ALLOCATION_MANAGER_ADDRESS=  # Address of the allocation manager
   ```

   ### Network Configuration
   ```
   FORK_URL=                    # URL of the RPC to fork from (Ethereum/Holesky)

>    RPC_URL=http://ethereum:8545 # Local RPC endpoint

Reviewer: ```suggestion
   RPC_URL=http://ethereum:8545 # Local RPC endpoint, change to a live endpoint when running on testnet
```

>    ENVIRONMENT=LOCAL            # Environment mode (LOCAL or TESTNET)

Reviewer: ```suggestion
   ENVIRONMENT=LOCAL            # Environment mode (LOCAL, MAINNET or TESTNET)
```

   FUNDED_KEY=                  # Private key used for funding test accounts (required for TESTNET)
   TEST_ACCOUNTS=3              # Number of test accounts to create
   ```

   ### Service Configuration
   ```
   CERBERUS_GRPC_PORT=50051     # Port for Cerberus gRPC service
   CERBERUS_METRICS_PORT=9081   # Port for Cerberus metrics
   SIGNER_ENDPOINT=http://signer:50051 # Endpoint for the signer service
   ```

   ### Optional Debug Configuration
   ```
   RUST_BACKTRACE=full          # Enable full backtrace for Rust errors
   RUST_LOG=debug              # Set Rust logging level
   ```

3. Build and start the services:
   ```
   # First, build the services with no cache
   docker-compose build --no-cache
   
   # Then start the services
   docker-compose up
   ```
Note that the nodes and node selector only start up after the eigenlayer setup container has exited  

## Running in TESTNET Mode

To run the service in TESTNET mode (Holesky), follow these steps:

1. Update your `.env` file with the following changes:

   a. Change the environment to TESTNET:
   ```
   ENVIRONMENT=TESTNET
   ```

   b. Set up your RPC URLs:
   ```

>    # Get a free RPC URL from providers like Infura, Alchemy, or QuickNode

Reviewer: ```suggestion
   # It's reccomended to get a paid RPC URL for non-flakey behavior
```

>    FORK_URL=https://holesky.infura.io/v3/YOUR-PROJECT-ID
>    RPC_URL=https://holesky.infura.io/v3/YOUR-PROJECT-ID

Reviewer: ```suggestion
   FORK_URL=https://holesky.rpc # Not a real RPC
   RPC_URL=https://holesky.rpc # Not a real RPC
```

   ```

>    c. Uncomment and use the Holesky configuration addresses:

Reviewer: ```suggestion
   c. Uncomment and use the testnet configuration addresses (The following are the addresses for the Holesky Testnet):
```

   ```
   DELEGATION_MANAGER_ADDRESS=0xA44151489861Fe9e3055d95adC98FbD462B948e7
   STRATEGY_MANAGER_ADDRESS=0xdfB5f6CE42aAA7830E94ECFCcAd411beF4d4D5b6
   LST_CONTRACT_ADDRESS=0x3F1c547b21f65e10480dE3ad8E19fAAC46C95034
   LST_STRATEGY_ADDRESS=0x7D704507b76571a51d9caE8AdDAbBFd0ba0e63d3
   BLS_SIGNATURE_CHECKER_ADDRESS=0xca249215e082e17c12bb3c4881839a3f883e5c6b
   OPERATOR_STATE_RETRIEVER_ADDRESS=0xB4baAfee917fb4449f5ec64804217bccE9f46C67
   ALLOCATION_MANAGER_ADDRESS=0x78469728304326CBc65f8f95FA756B0B73164462
   ```

   d. Set up your testnet keys:
   ```

>    # Generate a new private key or use an existing one with testnet ETH
>    PRIVATE_KEY=your_private_key_here
>    FUNDED_KEY=your_funded_private_key_here

Reviewer: ```suggestion
   # Generate a new private key or use an existing one with testnet ETH (the funded key will be used to transfer money to the PRIVATE_KEY account)
   PRIVATE_KEY=your_private_key_here
   FUNDED_KEY=your_funded_private_key_here
```

   ```

2. Get Testnet ETH:
   - Visit the [Holesky Faucet](https://holesky-faucet.pk910.de/) to get testnet ETH
   - Send some ETH to both your `PRIVATE_KEY` and `FUNDED_KEY` addresses
   - You can check your balance using [Holesky Etherscan](https://holesky.etherscan.io/)

> 3. Build and start the services:
>    ```
>    # First, build the services with no cache
>    docker-compose build --no-cache
>    
>    # Then start the services
>    docker-compose up
>    ```

Reviewer: ```suggestion
3. Build and start the services (assuming you have already built the images):
   ```
   #  Start the services
   docker-compose up
   ```
```

Author: resolved with https://github.com/BreadchainCoop/eigenlayer-bls-local/pull/14/commits/d1397d8b0ccfb57d9cbc5b740bca5526cd1a5281
